lambda-functions/cleanup-unused-images/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	log "github.com/sirupsen/logrus"
+)
+
+func main() {
+	log.SetLevel(log.DebugLevel)
+
+	lambda.Start(handler)
+}
+
+func handler() {
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(os.Getenv("AWS_REGION"))},
+	)
+
+	svc := ec2.New(sess)
+	uniqueUsedImages, err := getUniqueUsedImages(svc)
+	err = deleteAMIs(svc,uniqueUsedImages)
+
+	fmt.Println(uniqueUsedImages)
+
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				log.WithError(aerr.Error("AWS Error."))
+			}
+		} else {
```

Comment:
You would only need to check the AWS error code type if you want to do multiple things depending on the error type. If you just want to log the error then there is no need to check this first.

Reply:
There are some cases that the `aerr` is more informative, but in general I agree, you are right.